<a href="https://colab.research.google.com/github/lajmej/Analisis_Numerico/blob/main/Tarea4/tarea4Ejercicio6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Ejercicio 6

**Integrantes:**



*   De la Luz Siles Cristobal
*   Mejia Quinarez Luis Eduardo


**Sea la función $f(x)=x^{4} + 2x^{2} -x -3$**

In [2]:
def FixPoint(g,x0,Tol=0.0001,N=100):
    n=1
    while n<=N:
        x1=g(x0)
        print("iteración numero " + str(n) + " resultado= "+ str(x1))
        print("Distancia |x_k - x_k-1|: " + str(abs(x1-x0)))
        if abs(x1-x0)<=Tol:
            print(f"EL punto fijo es {x1}")
            return x1
        x0=x1
        n+=1
    else:
        print(f"Una aproximación del punto fijo es {x1}")
        return x1





**a) Manipulando la función $f$ da dos versiones de ella de tal forma que se pueda ver como $f(x)=g_{i}(x) -x$ de punto fijo para $i=1,2$**

*$f(x)=g_{1}-x=\sqrt{ \frac{x+3-x^{4}}{2}} -x$*

*$f(x)=g_{2}-x=\sqrt[4]{ x+3-x^{4}} -x$*

**b) De cada una de la iteraciones realizar 4 iteraciones del método de Punto Fijo con $x_{0}=1$.**

In [3]:
g_1= lambda x: ((x + 3 - (x**4))/2.0)**(0.5)

FP=FixPoint(g_1,1.0,0.00001,4)

iteración numero 1 resultado= 1.224744871391589
Distancia |x_k - x_k-1|: 0.22474487139158894
iteración numero 2 resultado= 0.9936661590774817
Distancia |x_k - x_k-1|: 0.23107871231410726
iteración numero 3 resultado= 1.228568645274987
Distancia |x_k - x_k-1|: 0.23490248619750542
iteración numero 4 resultado= 0.9875064291508866
Distancia |x_k - x_k-1|: 0.2410622161241005
Una aproximación del punto fijo es 0.9875064291508866




*   $g_{1}=\sqrt{\frac{x+3-x^{4}}{2}}$
*   $x_{0}=1$
*   $tolerancia=\frac{1}{1000}$
*   Limite de iteraciones $=4$

*Primera iteración: iteracion=1*
---

---


**$x_{0}$=1**

**n=1**


**Mientras $n=1 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=1.2247$**



**Si $|x_{1}-x_{0}|=0.2247\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=1.2247,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=2$**

*Segunda iteración: iteracion=2*


---



**$x_{0}$=1.2247**

**n=2**


**Mientras $n=2 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=0.9815$**



**Si $|x_{1}-x_{0}|=0.0184\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=0.9815,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=3$**

*Tercera iteración: iteracion=3*


---



**$x_{0}$=0.9815**

**n=3**


**Mientras $n=3 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=1.2355$**



**Si $|x_{1}-x_{0}|=0.2355\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=1.2355,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=4$**

*Cuarta iteración: iteracion=4*


---



**$x_{0}$=1.2355**

**n=4**


**Mientras $n=4 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=0.9754$**



**Si $|x_{1}-x_{0}|=0.0240\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=0.9754,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=4$**





In [4]:
g_2=lambda x: (x + 3 - 2*(x**2))**(1/4)

FP=FixPoint(g_2,1.0,0.00001,4)

iteración numero 1 resultado= 1.189207115002721
Distancia |x_k - x_k-1|: 0.18920711500272103
iteración numero 2 resultado= 1.0800577526675623
Distancia |x_k - x_k-1|: 0.1091493623351587
iteración numero 3 resultado= 1.1496714305893827
Distancia |x_k - x_k-1|: 0.06961367792182038
iteración numero 4 resultado= 1.1078205295102599
Distancia |x_k - x_k-1|: 0.041850901079122815
Una aproximación del punto fijo es 1.1078205295102599




*   $g_{2}=\sqrt[4]{x+3-2x^{2}}$
*   $x_{0}=1$
*   $tolerancia=\frac{1}{1000}$
*   Limite de iteraciones $=4$

*Primera iteración: iteracion=1*
---

---


**$x_{0}$=1**

**n=1**


**Mientras $n=1 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=1.2038$**



**Si $|x_{1}-x_{0}|=0.2038\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=1.2038,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=2$**

*Segunda iteración: iteracion=2*


---



**$x_{0}$=1.2038**

**n=2**


**Mientras $n=2 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=1.0688$**



**Si $|x_{1}-x_{0}|=0.0688\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=1.0688,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=3$**

*Tercera iteración: iteracion=3*


---



**$x_{0}$=1.0688**

**n=3**


**Mientras $n=3 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=1.1556$**



**Si $|x_{1}-x_{0}|=0.1556\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=1.1556,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=4$**

*Cuarta iteración: iteracion=4*


---



**$x_{0}$=1.1556**

**n=4**


**Mientras $n=4 \leq 4$:**

**$x_{1}=g_{1}(x_{0})=1.1038$**



**Si $|x_{1}-x_{0}|=0.1038\leq \frac{1}{1000}$**

**entonces regresa: $x_{1}=1.1038,$**
**pero no se cumple.**


**$x_{0}=x_{1}$**

**$n=n+1=4$**



